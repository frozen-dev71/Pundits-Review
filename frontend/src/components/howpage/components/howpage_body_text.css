/*
// ======================================
// CSS for How Page Body Text Component 
// ======================================
*/


.how-page-body-text-wrapper {
	display: flex;
	width: 80%;
	align-items: center;
	justify-content: center;
}


.how-page-body-text {
	font-size: .85em;
	letter-spacing: .05em;
	text-align: center;
}



/*
// ======================================
// MEDIA QUERIES
// ======================================
*/

@media (min-width: 600px) {


	.how-page-body-text-wrapper {
	
		width: 70%;

	}

	.how-page-body-text {
		font-size: 1em;
		padding: .5em 0;
	}

}


@media (min-width: 900px) {

	.how-page-body-text-wrapper {
	
		width: 60%;

	}

	.how-page-body-text {
		font-size: 1.15em;
		padding: 1em 0;
	}

}



@media (min-width: 1200px) {

	.how-page-body-text-wrapper {
	
		width: 50%;

	}

	.how-page-body-text {
		padding: 1.5em 0;
	}

}