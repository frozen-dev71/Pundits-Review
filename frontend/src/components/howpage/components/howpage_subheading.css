/*
// ======================================
// CSS for How It Works Sub Heading Element
// ======================================
*/

.how-page-subheading-wrapper {
	display: flex;
	align-items: center;
	padding: 1em 0;
	margin-top: 1em;
}


.how-page-subheading-text {

	letter-spacing: .1em;
	font-size: 1.25em;
}




/*
// ======================================
// MEDIA QUERIES
// ======================================
*/

@media (min-width: 600px) {

	.how-page-subheading-wrapper {
		padding: 2em 0;
	}

	.how-page-subheading-text {
		font-size: 1.5em;
	}

}


@media (min-width: 900px) {

	.how-page-subheading-wrapper {
		padding: 3em 0;
	}

	.how-page-subheading-text {
		font-size: 2em;
	}

}