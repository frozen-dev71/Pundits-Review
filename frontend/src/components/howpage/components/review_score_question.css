/*
// ======================================
// CSS for Review Score Question Wrapper - How Page
// ======================================
*/


.review-score-question-wrapper {
	display: flex;
	width: 70%;
	justify-content: center;
	align-items: center;
    padding-top: 2em;
}

.arrow-question-image {
    width: 100%;
    height: 15em;
    background-size: contain;
    background-repeat: no-repeat;
    background-position: center;
    background-image: url(./../resources/top-image.png);
}





/*
// ======================================
// MEDIA QUERIES
// ======================================
*/

@media (min-width: 600px) {

    .review-score-question-wrapper {
        padding-top: 4em;
    }

}



@media (min-width: 900px) {


    .arrow-question-image {
        height: 20em;
    }

}