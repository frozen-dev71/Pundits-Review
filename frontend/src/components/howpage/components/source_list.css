/*
// ======================================
// CSS for Source List Component 
// ======================================
*/

.source-list-wrapper {
	display: none;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-top: 3em;
}

.how-page-source-item {
	letter-spacing: .1em;
	font-size: .9em;
	padding: 2em 0;
	text-align: center;
	color: black;
	cursor: pointer;

}

.source-item-wrapper {
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
}

