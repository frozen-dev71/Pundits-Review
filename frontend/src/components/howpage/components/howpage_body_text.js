// ======================================
// How Page Body Text Component - contains text for the top par & text to explain each subheading
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";

// CSS Imports 
import './howpage_body_text.css';



// ===========
// How Page Body Text Component 
// ===========

export class HowPageBodyText extends Component {

	render() {

		let result = (

				<div className="how-page-body-text-wrapper">

					<p className="how-page-body-text body-text-light">{this.props.text}</p>

				</div>

			);


		return result;
	}
}