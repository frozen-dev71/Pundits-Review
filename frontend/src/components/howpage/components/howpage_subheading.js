// ======================================
// How Page Sub Heading Component - contains subheading text
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";

// CSS Imports 
import './howpage_subheading.css';


// ===========
// How Page SubHeading Component 
// ===========


export class HowPageSubHeading extends Component {

	render() {

		let result = (

				<div className="how-page-subheading-wrapper">

					<p className="how-page-subheading-text heading-text">{this.props.text}</p>

				</div>

			);


		return result;
	}
}