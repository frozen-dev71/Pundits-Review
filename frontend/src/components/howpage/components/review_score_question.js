// ======================================
// Review Score Question Component - Contains review score circle and illustration 
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";


// Local JS Imports



// CSS Imports 
import './review_score_question.css';


// ===========
// Review Score Question Component 
// ===========

export class ReviewScoreQuestion extends Component {

	render() {

		let result = (

				<div className="review-score-question-wrapper">

					<span className="arrow-question-image"></span>

				</div> 


			);


		return result;
	}
}