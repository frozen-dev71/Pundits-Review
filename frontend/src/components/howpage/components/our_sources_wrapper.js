// ======================================
// Our Sources Wrapper Component - contains elements displaying the sources used in pundits review system
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";

// Local JS Imports
import { OurSourcesBackground } from './our_sources_background_image';
import { getSourcesList, clearSourcesList } from '../../actions/actions';
import { SourceList } from './source_list';

// CSS Imports 
import './our_sources.css';

// ===========
// Our Sources Wrapper Component 
// ===========

export class OurSources extends Component {

	componentDidMount = () => {

		this.props.getSourcesList();

	}

	_handleClick = () => {

		document.querySelector('.source-list-wrapper').style.display = "flex";
	}


	render() {

		let result = (

				<div className="our-sources-wrapper">


					<p className="body-text-light our-sources-title">OUR SOURCES</p>

					<p className="body-text-light our-sources-subtitle">Our information is collected from digital football media publishers. We avoid fansites, blogs and official club pages where possible but are always open to new suggestions.</p>

					<button className="view-sources-button body-text-light" onClick={this._handleClick}>
						VIEW OUR SOURCES
					</button>

					<SourceList />

				</div>

			);

		return result;
	}
}


// ===============
// Connect redux state container to search component
// ===============

const mapStateToProps = state => {
  return { 
	  	sourcesList: state.sourcesList,
  	};
};


// Export connected component
export const OurSourcesWrapper = connect(mapStateToProps, { getSourcesList, clearSourcesList })(OurSources);


