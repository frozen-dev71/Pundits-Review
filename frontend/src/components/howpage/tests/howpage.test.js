// ======================================
// This file tests for the components listed in the GLOBAL sub-directory
// ======================================


// ===========
// TEST COVERAGE
// ===========

// 1. Check that HowPage renders correctly
// 2. Check that HowPageBodyText renders correctly
// 3. Check that HowPageSubHeading renders correctly
// 4. Check that OurSourcesWrapper renders correctly
// 5. Check that ReviewScoreQuestion renders correctly
// 6. Check that SourceList renders correctly

// ===========
// IMPORTS 
// ===========


// General Imports
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';

// Local JS Imports
import { HowPage } from '.././howpage';
import { HowPageBodyText } from '.././components/howpage_body_text';
import { HowPageSubHeading } from '.././components/howpage_subheading';
import { OurSourcesWrapper } from '.././components/our_sources_wrapper';
import { ReviewScoreQuestion } from '.././components/review_score_question';
import { SourceList } from '.././components/source_list';


import store from '../../store/store';

// Configurations
Enzyme.configure({ adapter: new Adapter() })

// ===========
// Suppression of SSL Warning 
//
// Code adapted from comment from sketchbuch on GitHub issues post 'Warning: useLayoutEffect does nothing on the server #1373' on 12 Oct 2019
// https://github.com/reduxjs/react-redux/issues/1373
// Accessed 09 Sep 2020
//
// __Note__:
//	
//		- Jest produces "Warning: useLayoutEffect does nothing on the server" & 'Warning: React.createElement: type is invalid -- expected a string (for built-in components)' - warning which confuses test cases 
// 		- Wrapper test suppresses that warning for clear test reports
// ===========

describe('HowPage Component Tests', () => {
  	const originalConsoleError = console.error;

  	beforeEach(() => {
    	console.error = jest.fn((msg) => {
	      	if (msg.includes('Warning: useLayoutEffect does nothing on the server') || msg.includes('Warning: React.createElement: type is invalid -- expected a string (for built-in components)')) {
	       		return null;
	      	} else {
	        	originalConsoleError(msg);
	      	}
    	});
  	});

  afterEach(() => {
    console.error = originalConsoleError;
  });


// End of referenced code

  	// ===========
	// TEST 1. Check that HowPage renders correctly
	// ===========

	describe('HowPage', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><HowPage/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 2. Check that HowPageBodyText renders correctly
	// ===========

	describe('HowPageBodyText', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><HowPageBodyText/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 3. Check that HowPageSubHeading renders correctly
	// ===========

	describe('HowPageSubHeading', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><HowPageSubHeading/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 4. Check that OurSourcesWrapper renders correctly
	// ===========

	describe('OurSourcesWrapper', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><OurSourcesWrapper/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 5. Check that ReviewScoreQuestion renders correctly
	// ===========

	describe('ReviewScoreQuestion', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ReviewScoreQuestion/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 6. Check that SourceList renders correctly
	// ===========

	describe('SourceList', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><SourceList/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


});
