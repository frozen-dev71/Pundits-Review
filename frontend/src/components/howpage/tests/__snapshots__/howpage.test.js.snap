// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowPage Component Tests HowPage should render correctly 1`] = `
<div
  class="how-page-wrapper"
>
  <p
    class="how-page-title heading-text"
  >
    How It Works
  </p>
  <div
    class="how-page-body-text-wrapper"
  >
    <p
      class="how-page-body-text body-text-light"
    >
      Every week, Pundits Review collects articles about the Premier League, analyses them and then gives a score to players and clubs based on how they were reviewed within them.
    </p>
  </div>
  <div
    class="review-score-question-wrapper"
  >
    <span
      class="arrow-question-image"
    />
  </div>
  <div
    class="how-section-wrapper"
  >
    <div
      class="how-page-subheading-wrapper"
    >
      <p
        class="how-page-subheading-text heading-text"
      >
        Collecting Content
      </p>
    </div>
    <div
      class="how-page-body-text-wrapper"
    >
      <p
        class="how-page-body-text body-text-light"
      >
        Our system makes weekly visits to over 60 football websites and finds new articles about the Premier League.
      </p>
    </div>
    <div
      class="how-page-image-container"
    >
      <span
        class="article-outline-image howpage-image"
      />
    </div>
  </div>
  <div
    class="dividing-line background-dark"
  />
  <div
    class="how-section-wrapper"
  >
    <div
      class="how-page-subheading-wrapper"
    >
      <p
        class="how-page-subheading-text heading-text"
      >
        Picking Out Phrases
      </p>
    </div>
    <div
      class="how-page-body-text-wrapper"
    >
      <p
        class="how-page-body-text body-text-light"
      >
        We collect the text from each new article we find, then divide it into short phrases.
      </p>
    </div>
    <div
      class="how-page-image-container"
    >
      <span
        class="article-phrases-image howpage-image"
      />
    </div>
  </div>
  <div
    class="dividing-line background-dark"
  />
  <div
    class="how-section-wrapper"
  >
    <div
      class="how-page-subheading-wrapper"
    >
      <p
        class="how-page-subheading-text heading-text"
      >
        Identifying Targets
      </p>
    </div>
    <div
      class="how-page-body-text-wrapper"
    >
      <p
        class="how-page-body-text body-text-light"
      >
        We process the article phrases one at a time, firstly looking out for which player or club is being written about.
      </p>
    </div>
    <div
      class="how-page-image-container-short"
    >
      <span
        class="target-identifier-image howpage-image"
      />
    </div>
  </div>
  <div
    class="dividing-line background-dark"
  />
  <div
    class="how-section-wrapper"
  >
    <div
      class="how-page-subheading-wrapper"
    >
      <p
        class="how-page-subheading-text heading-text"
      >
        Predicting Sentiment
      </p>
    </div>
    <div
      class="how-page-body-text-wrapper"
    >
      <p
        class="how-page-body-text body-text-light"
      >
        Based on the other words in the phrase, our model predicts whether it is a positive, negative or neutral review. You might not always agree with our predictions!
      </p>
    </div>
    <div
      class="how-page-image-container"
    >
      <span
        class="predict-sentiment-image howpage-image"
      />
    </div>
  </div>
  <div
    class="dividing-line background-dark"
  />
  <div
    class="how-section-wrapper"
  >
    <div
      class="how-page-subheading-wrapper"
    >
      <p
        class="how-page-subheading-text heading-text"
      >
        Calculating Review Scores
      </p>
    </div>
    <div
      class="how-page-body-text-wrapper"
    >
      <p
        class="how-page-body-text body-text-light"
      >
        After every phrase has been processed, we calculate review scores for players and clubs.
      </p>
    </div>
    <div
      class="how-page-image-container"
    >
      <span
        class="calculate-score-image howpage-image"
      />
    </div>
  </div>
  <div
    class="dividing-line background-dark"
  />
  <div
    class="our-sources-wrapper"
  >
    <p
      class="body-text-light our-sources-title"
    >
      OUR SOURCES
    </p>
    <p
      class="body-text-light our-sources-subtitle"
    >
      Our information is collected from digital football media publishers. We avoid fansites, blogs and official club pages where possible but are always open to new suggestions.
    </p>
    <button
      class="view-sources-button body-text-light"
    >
      VIEW OUR SOURCES
    </button>
    <span />
  </div>
  <div
    class="footer-wrapper"
  >
    <p
      class="footer-title body-text-light"
    >
      GET IN TOUCH
    </p>
    <div
      class="footer-logo-wrapper"
    >
      <a
        class="react-link"
        href="https://twitter.com/AndyClarkeMedia"
        target="_blank"
      >
        <i
          aria-hidden="true"
          class="fa fa-twitter media-icon"
        />
      </a>
      <a
        class="footer-logo heading-text react-link"
        href="/"
      >
        PR
      </a>
      <a
        class="react-link"
        href="mailto:clarkeAJ3@cardiff.ac.uk"
        target="_blank"
      >
        <i
          aria-hidden="true"
          class="fa fa-envelope media-icon"
        />
      </a>
    </div>
    <p
      class="footer-copyright-notice body-text-light"
    >
      Â© Andy Clarke, 2020
    </p>
  </div>
</div>
`;

exports[`HowPage Component Tests HowPageBodyText should render correctly 1`] = `
<div
  class="how-page-body-text-wrapper"
>
  <p
    class="how-page-body-text body-text-light"
  />
</div>
`;

exports[`HowPage Component Tests HowPageSubHeading should render correctly 1`] = `
<div
  class="how-page-subheading-wrapper"
>
  <p
    class="how-page-subheading-text heading-text"
  />
</div>
`;

exports[`HowPage Component Tests OurSourcesWrapper should render correctly 1`] = `
<div
  class="our-sources-wrapper"
>
  <p
    class="body-text-light our-sources-title"
  >
    OUR SOURCES
  </p>
  <p
    class="body-text-light our-sources-subtitle"
  >
    Our information is collected from digital football media publishers. We avoid fansites, blogs and official club pages where possible but are always open to new suggestions.
  </p>
  <button
    class="view-sources-button body-text-light"
  >
    VIEW OUR SOURCES
  </button>
  <span />
</div>
`;

exports[`HowPage Component Tests ReviewScoreQuestion should render correctly 1`] = `
<div
  class="review-score-question-wrapper"
>
  <span
    class="arrow-question-image"
  />
</div>
`;

exports[`HowPage Component Tests SourceList should render correctly 1`] = `<span />`;
